tools/src/test/java/com/nedap/archie/flattener/InMemoryFullArchetypeRepositoryTest.java

```diff
@@ -0,0 +1,121 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.*;
+
+public class InMemoryFullArchetypeRepositoryTest {
+
+    InMemoryFullArchetypeRepository inMemoryFullArchetypeRepository;
+
+    @Before
+    public void setup() throws IOException {
+        //Create versioned archetypes
+        List<Archetype> archetypes = new ArrayList<>();
+        archetypes.add(new ADLParser().parse(
+                InMemoryFullArchetypeRepositoryTest.class.
+                        getResourceAsStream("openEHR-EHR-OBSERVATION.blood_pressure.v1.adls")));
+        archetypes.add(new ADLParser().parse(
+                InMemoryFullArchetypeRepositoryTest.class.
+                        getResourceAsStream("openEHR-EHR-OBSERVATION.blood_pressure.v1.0.1.adls")));
+        archetypes.add(new ADLParser().parse(
+                InMemoryFullArchetypeRepositoryTest.class.
+                        getResourceAsStream("openEHR-EHR-OBSERVATION.blood_pressure.v1.0.11.adls")));
+        archetypes.add(new ADLParser().parse(
+                InMemoryFullArchetypeRepositoryTest.class.
+                        getResourceAsStream("openEHR-EHR-OBSERVATION.blood_pressure.v1.1.0.adls")));
+        archetypes.add(new ADLParser().parse(
+                InMemoryFullArchetypeRepositoryTest.class.
+                        getResourceAsStream("openEHR-EHR-OBSERVATION.blood_pressure.v1.11.0.adls")));
+        archetypes.add(new ADLParser().parse(
+                InMemoryFullArchetypeRepositoryTest.class.
+                        getResourceAsStream("openEHR-EHR-OBSERVATION.blood_pressure.v2.0.0.adls")));
+
+        //Create ValidationResults from versioned archetypes
+        List<ValidationResult> validationResults = archetypes.
+                stream().
+                map(ValidationResult::new).
+                collect(Collectors.toList());
+
+        //Create OperationalTemplates
+        List<OperationalTemplate> operationalTemplates = archetypes.
+                stream().
+                map(OperationalTemplateCreator::createOperationalTemplate).
+                collect(Collectors.toList());
+
+        //Fill InMemoryFullArchetypeRepository
+        inMemoryFullArchetypeRepository = new InMemoryFullArchetypeRepository();
+        for (int i = 0; i < archetypes.size(); ++i) {
+            inMemoryFullArchetypeRepository.addArchetype(archetypes.get(i));
+            inMemoryFullArchetypeRepository.setFlattenedArchetype(archetypes.get(i));
+            inMemoryFullArchetypeRepository.setValidationResult(validationResults.get(i));
+            inMemoryFullArchetypeRepository.setOperationalTemplate(operationalTemplates.get(i));
+        }
+    }
+
+    @Test
+    public void getArchetype() {
+        assertEquals("openEHR-EHR-OBSERVATION.blood_pressure.v2.0.0",
+                inMemoryFullArchetypeRepository.getArchetype("openEHR-EHR-OBSERVATION.blood_pressure.v").getArchetypeId().getFullId());
```

Comment:
In the spec I think a version number of at least 1 digit is always required. But maybe good to support a lookup of the latest version of an archetype regardless of a version. In that case, should we include the `.v` or leave it out altogether?

Reply:
Removed the `.v` for that case. This required some more logic in the `ArchetypeHRID` (and similarly in the `ArchetypeID`). Maybe we should think of adding fields for fullID, semanticID, etc. at some point?